src/NHibernate/Mapping/ByCode/ModelMapper.cs

```diff
@@ -1327,19 +1327,19 @@ private IMapKeyRelationMapper DetermineMapKeyRelationType(MemberInfo member, Pro
 
 		private class ComponentRelationMapper : ICollectionElementRelationMapper
 		{
-			private readonly MemberInfo collectionMember;
+			private readonly PropertyPath propertyPath;
 			private readonly System.Type componentType;
 			private readonly ICustomizersHolder customizersHolder;
 			private readonly IModelInspector domainInspector;
 			private readonly ICandidatePersistentMembersProvider membersProvider;
 			private readonly ModelMapper modelMapper;
 			private readonly System.Type ownerType;
 
-			public ComponentRelationMapper(MemberInfo collectionMember, System.Type ownerType, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
+			public ComponentRelationMapper(PropertyPath propertyPath, System.Type componentType, ICandidatePersistentMembersProvider membersProvider,
 			                               IModelInspector domainInspector, ICustomizersHolder customizersHolder, ModelMapper modelMapper)
 			{
-				this.collectionMember = collectionMember;
-				this.ownerType = ownerType;
+				this.propertyPath = propertyPath;
+				this.ownerType = propertyPath.LocalMember.DeclaringType;
```

Comment:
I have figured out where `propertyPath.DeclaringMember.OwnerType` differs from the `ownerType`, and that is where you are mapping inherited members.
I've changed the code so it now passes the ownerType explicitly